odl/test/operator/operator_test.py

```diff
@@ -47,33 +57,46 @@ def __str__(self):
         return "MaS: " + str(self.matrix) + " ** 2"
 
 
-def mult_sq_np(A, x):
-    # The same as MultiplyAndSquareOp but only using numpy
-    return np.dot(A, x) ** 2
+def mult_sq_np(mat, x):
+    """NumPy reference implementation of MultiplyAndSquareOp."""
+    return np.dot(mat, x) ** 2
 
 
-def test_nonlinear_op():
-    # Verify that the operator does indeed work as expected
-    A = np.random.rand(4, 3)
-    x = np.random.rand(3)
-    Aop = MultiplyAndSquareOp(A)
-    xvec = Aop.domain.element(x)
+def check_call(operator, x, expected):
+    """Assert that operator(point) == expected."""
+    # Out-of-place check
+    assert all_almost_equal(operator(x), expected)
 
-    assert all_almost_equal(Aop(xvec), mult_sq_np(A, x))
+    # In-place check, no aliasing
+    out = operator.range.element()
+    operator(x, out=out)
+    assert all_almost_equal(out, expected)
 
+    # In-place check, aliased
+    if operator.domain == operator.range:
+        y = x.copy()
+        operator(y, out=y)
+        assert all_almost_equal(y, expected)
 
-def check_call(operator, point, expected):
-    """Assert that operator(point) == expected."""
 
-    assert all_almost_equal(operator(point), expected)
+# --- Unit tests --- #
 
-    out = operator.range.element()
-    operator(point, out=out)
 
-    assert all_almost_equal(out, expected)
+def test_operator_call(dom_eq_ran):
+    """Check operator evaluation against NumPy reference."""
+    if dom_eq_ran:
```

Comment:
prefer adding a "shape" fixture?

Reply:
That would make it quite complicated, e.g., when testing `op1 + op2` shapes must be the same, when testing `op1 * op2` you have another requirement, etc. This fixture is really only intended to generate a test case where aliased input and output are checked.

Reply:
I can add a comment in the test.

Reply:
:+1: